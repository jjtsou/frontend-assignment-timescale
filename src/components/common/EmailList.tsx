import { ReactNode } from 'react';
import Collapse from './Collapse';

type EmailListTypes = {
  domain: string;
  openByDefault?: boolean;
  onSelectDomain?: () => void;
  children: ReactNode;
};

const EmailList = ({
  domain,
  onSelectDomain,
  children,
}: EmailListTypes): JSX.Element => (
  <Collapse header={domain} onClickHeader={onSelectDomain}>
    {/* {emails.map(({ email, isSelected }, i) => {
      return (
        <EmailListItem
          key={`${email}_${i}`}
          isSelected={isSelected}
          onClick={
            selectRecipient
              ? () => selectRecipient?.({ email, domain, isSelected })
              : undefined
          }
          value={email}
        />
      );
    })} */}
    {children}
  </Collapse>
);

export default EmailList;
