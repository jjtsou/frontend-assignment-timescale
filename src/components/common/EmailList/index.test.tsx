import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import EmailList from '../EmailList';

describe('EmailList component', () => {
  const domain = 'timescale.com';
  it('should render properly', () => {
    const { container } = render(
      <EmailList domain={domain}>
        <span>List item</span>
      </EmailList>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with group', () => {
    const { container } = render(
      <EmailList domain={domain} isGroup>
        <span>List item</span>
      </EmailList>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with openByDefault', () => {
    const { container } = render(
      <EmailList domain={domain} isGroup openByDefault>
        <span>List item</span>
      </EmailList>
    );
    expect(container).toMatchSnapshot();
  });

  it('should call onSelectDomain when header is clicked', async () => {
    const onSelectDomain = jest.fn();
    const { getByText } = render(
      <EmailList domain={domain} isGroup onSelectDomain={onSelectDomain}>
        <span>List item</span>
      </EmailList>
    );
    const header = getByText(domain);
    await userEvent.click(header);
    expect(onSelectDomain).toHaveBeenCalled();
  });
});
