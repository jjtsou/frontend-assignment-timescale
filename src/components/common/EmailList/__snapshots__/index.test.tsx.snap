// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailList component should render properly 1`] = `
<div>
  <span>
    List item
  </span>
</div>
`;

exports[`EmailList component should render with group 1`] = `
<div>
  <div
    class="css-j7qwjs"
  >
    <div
      class="css-1m29ivc"
    >
      <button
        aria-label="Open collapsible panel"
        class="chakra-button css-9imh88"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            fill="currentColor"
          />
        </svg>
      </button>
      <p
        class="chakra-text chakra-text css-1efywoh"
      >
        timescale.com
      </p>
    </div>
    <div
      class="chakra-collapse"
      style="overflow: hidden; display: none; opacity: 0; height: 0px;"
    >
      <div
        class="css-11qo4fr"
      >
        <span>
          List item
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`EmailList component should render with openByDefault 1`] = `
<div>
  <div
    class="css-j7qwjs"
  >
    <div
      class="css-1m29ivc"
    >
      <button
        aria-label="Open collapsible panel"
        class="chakra-button css-9imh88"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </button>
      <p
        class="chakra-text chakra-text css-1efywoh"
      >
        timescale.com
      </p>
    </div>
    <div
      class="chakra-collapse"
      style="overflow: hidden; display: block; opacity: 1; height: auto;"
    >
      <div
        class="css-11qo4fr"
      >
        <span>
          List item
        </span>
      </div>
    </div>
  </div>
</div>
`;
